### DISCUSSION TITLE
libusb.h: No such file or directory

### DISCUSSION THREAD

**Discussion by huojiujian1** [jankae/LibreVNA#264] [Q&A]
![1](https://github.com/user-attachments/assets/a2495b38-d633-4678-8bf1-c735eb7b6d2a)
When attempting to compile Qt open source, I encountered the error 'libusb.h: No such file or directory.' Despite following methods found online to add the library, I still cannot successfully compile it.

**Comment by jankae** [comment]
Without knowing what exactly you did, I can not help you here. The message simply tells me that your compiler can not find the libusb header. Make sure to either copy this file into the include path of your compiler or add the location of the file to the search path for includes.

**Reply by huojiujian1** [reply]
I just downloaded the source code of LibreVNA-GUI-1.6.0 and put it into Qt to open the LibreVNA.pro project to run, and then the error libusb.h: No such file or directory.
![image](https://github.com/user-attachments/assets/0d55ad28-c98f-4c40-986a-b7c2d2e7a678)

**Reply by jankae** [reply]
Did you install the libusb library? That is a separate dependency and not included in the LibreVNA code?

**Reply by huojiujian1** [reply]
I installed the libusb library from github, how do I add or use the library to the project?

**Reply by jankae** [reply]
This really depends on where your compiler is installed (i.e. where it is looking for header files by default) and how your include paths are configured.

May I ask why you are trying to compile the GUI yourself? If you just want to run it, there are binaries available.

Of course you might want to modify but in that case installing a dependency (which is a fairly standard procedure in a C++ project/toolchain) will be one of the smallest problems you have to solve.

**Reply by huojiujian1** [reply]
I just want to learn and use this LibreVNA and GUI, so I want to try compiling and running the source code. Thank you very much for your help. I apologize for any inconvenience.

**Reply by jankae** [reply]
If you just want to learn about the VNA and the GUI, I suggest you use the precompiled binaries. That way, there is no need to compile it yourself.
You really only need to install the toolchain and compile yourself when you want to add your own features to the GUI.

**Reply by huojiujian1** [reply]
I sincerely appreciate your valuable advice and regret any inconvenience my inquiry may have caused.

**Reply by jankae** [reply]
no need to apologize and I would have preferred helping you more but without knowing what exactly you installed (compiler, Qt version, libusb location,...) this is very difficult. Unfortunately, integrating external libraries in C/C++ is not as easy as e.g. in Python

**Reply by huojiujian1** [reply]
My compiler and QT version are Desktop Qt 6.5.3 MinGW 64-bit. The location of the libusb library is on the desktop. Also, I saw the code for adding the libusb library in the.pro file. After adding the path where my libusb library is located, namely C:$\Users\1123g\Desktop\libusb-1.0.27\MinGW64, I found that compilation failed. 
![image](https://github.com/user-attachments/assets/e873c89f-0d7d-48a1-af3a-abc2abf58879)
![image](https://github.com/user-attachments/assets/a8e972d0-13d1-47d0-a40b-f45a6a093b55)
![image](https://github.com/user-attachments/assets/58b1dde3-50aa-4283-8b08-996c10f4895e)

**Reply by jankae** [reply]
Can you post a screenshot of the error message and your modified .pro file as well? Not sure how familiar you are with C++ toolchains but there are two distinct steps: compiling and linking. Your "libusb.h: no such file or directory" is a compiler error. To fix it, you need to point your compiler to the directory of the libusb.h file.
What you are doing with the path to the library is only influencing the linker. You likely will have to do this anyway (otherwise you will get a linker error next) but at this point it will have no influence whatsoever on your compiler error.

With you setup, I would put the libusb.h here in a first attempt:
C:/Qt/6.5.3/mingw_64/include/libusb-1.0/libusb.h
This is a bit ugly (it requires you to repeat this when you update to new Qt versions) but probably the easiest way to solve your compiler error.

**Reply by huojiujian1** [reply]
![image](https://github.com/user-attachments/assets/c3cba361-9987-4339-a032-192ca838b697)
![image](https://github.com/user-attachments/assets/9fee7226-6665-4be2-84cb-6cb7052ab3b0)

**Reply by jankae** [reply]
Well, you have the include path shown right there. Looks a bit weird with the Qt6.7 followed by Qt6.5.3 but if that is your include path maybe try putting the libusb.h there?
The compiler is looking for "libusb-1.0/libusb.h" relative to your include path, so the file should be placed like this:
D:/Qt6.7/Qt6.5.3/mingw_64/include/libusb-1.0/libusb.h
